### ISSUE TITLE
Getting this to work with multiple projects with pages?

### ISSUE THREAD

**Issue by RonSijm** [skclusive/Skclusive.Material.Component#76]
Hey there,

I'm trying to get the latest version of this library to work, but I'm running in some issues with multiple projects that have pages.

Normally I'd do this by adding a `Router` tag to the main layout. 

````    
<Router AppAssembly="typeof(DashboardView).Assembly"
            AdditionalAssemblies="new[] { typeof(AccountStyles).Assembly }">
    
        <Found Context="routeData">
            <RouteView RouteData="@routeData"
                       DefaultLayout="@typeof(FullLayout)" />
        </Found>
        <NotFound>
            <LayoutView Layout="@typeof(FullLayout)">
                <p>Sorry, there's nothing at this address.</p>
            </LayoutView>
        </NotFound>
    </Router>
````

I downloaded all the samples and tried to get it to work in, for example the `Skclusive.Blazor.Dashboard` project, by moving the Account pages to a different project and then using that config above, but so far I can't figure out how to get it to work.

I noticed that in none of the sample projects there's a Router, and instead there is a `ThemeProvider`
 with a `RouterLayout` (I'm not sure if that is related)
````
<ThemeProvider
    Light="@Light"
    Dark="@Dark">
    <RouterLayout
        Main="@typeof(FullLayout)"
        Mini="@typeof(SmallLayout)" />
</ThemeProvider>
````

 ... but it only seems to work when I only have 1 assembly in which I put the pages. I'm not sure what I'm doing wrong.

I attached a project where I split up the demo project for "trying to get it to work" purposes
[Skclusive.Blazor.Dashboard.With.Seperated.View.zip](https://github.com/skclusive/Skclusive.Material.Component/files/5580498/Skclusive.Blazor.Dashboard.With.Seperated.View.zip)

**Comment by skclusive** [comment]
thanks for raising the issue. There is IRouteAssemblyProvider which can be used. just noticed missed the piece to push.

for now you can use Router as you do instead of RouterLayout wrapped by ThemeProvider as below.

```html
<ThemeProvider
    Light="@Light"
    Dark="@Dark">
   <Baseline />
   <Router
       AppAssembly="typeof(DashboardView).Assembly"
       AdditionalAssemblies="new[] { typeof(AccountStyles).Assembly }">
      <Found
          Context="routeData">
         <RouteView
              RouteData="@routeData"
              DefaultLayout="@typeof(FullLayout)" />
      </Found>
     <NotFound>
          <LayoutView
              Layout="@typeof(SmallLayout)">
               <NotFound />
          </LayoutView>
      </NotFound>
  </Router>
</ThemeProvider>
```

will add support for additional assembly param support for RouterLayout.

**Comment by RonSijm** [comment]
Ah thanks for the quick response. I was not aware that I could place the router inside the theme provider. And when it's placed on the outside weird things are happening

**Comment by skclusive** [comment]
fixed in 5.2.0 release. AdditionalAssemblies can be passed as below.

```html
<ThemeProvider
    Light="@Light"
    Dark="@Dark">
    <RouterLayout
       AppAssembly="typeof(DashboardView).Assembly"
       AdditionalAssemblies="new[] { typeof(AccountStyles).Assembly }"
        Main="@typeof(FullLayout)"
        Mini="@typeof(SmallLayout)" />
</ThemeProvider>

```

**Comment by skclusive** [comment]
closing the issue now. feel free to re-open if required.
